src/target/arm.rs

```diff
@@ -0,0 +1,295 @@
+            match o.vector_size() {
+                Some(vector_size) => {
+                    let volume = o.dim_sizes().iter().product::<DimSize>();
+                    if vector_size != volume {
+                        return false;
+                    }
+                }
+                None => {
+                    panic!("No vector_size on operand in level {:?}", o.level());
+                }
+            }
+        }
+    }
+    has_vrf
+}
+
+pub fn cacheaccess_applies_to_operands(operands: &[TensorSpec<ArmTarget>]) -> bool {
+    return false;
+
+    if operands.iter().all(|o| o.level().is_addressed()) {
+        return false;
+    }
+    if operands.iter().any(|o| !o.is_contiguous()) {
+        return false;
+    }
+    if operands[0].dtype() != operands[1].dtype() {
+        return false;
+    }
+    if operands[0].dim_sizes() != operands[1].dim_sizes() {
+        return false;
+    }
+    if operands[0].layout() != operands[1].layout() {
+        return false;
+    }
+    true
+}
+
+pub fn memsetzero_applies_to_operands(operands: &[TensorSpec<ArmTarget>]) -> bool {
+    if !operands[0].is_contiguous() {
+        return false;
+    }
+    if operands[0].level() != X86MemoryLevel::RF {
+        return false;
+    }
+    true
+}
+
+pub fn vectorzero_applies_to_operands(operands: &[TensorSpec<ArmTarget>]) -> bool {
+    if !operands[0].is_contiguous() {
+        return false;
+    }
+    if operands[0].level() != X86MemoryLevel::VRF {
+        return false;
+    }
+    let volume = operands[0].dim_sizes().iter().product::<DimSize>();
+    match operands[0].vector_size() {
+        Some(vector_size) if vector_size != volume => {
+            return false;
+        }
+        None => return false,
+        _ => (),
+    };
+    true
+}
+
+pub fn broadcastvecmult_applies_to_operands(operands: &[TensorSpec<ArmTarget>]) -> bool {
+    if operands[0].level() != X86MemoryLevel::RF {
+        return false;
+    }
+    for i in 1..3 {
+        if operands[i].level() != X86MemoryLevel::VRF {
+            return false;
+        }
+        let volume = operands[i].dim_sizes().iter().product::<DimSize>();
+        if volume != operands[i].vector_size().unwrap() {
+            return false;
+        }
+        if !operands[i].aligned() || !operands[i].is_contiguous() {
+            return false;
+        }
+        if operands[0].dtype() != operands[i].dtype() {
+            return false;
+        }
+    }
+    if operands[0].dim_sizes().iter().any(|d| *d != 1) {
+        return false;
+    }
+    if operands[1].dim_sizes().len() != 2 || operands[1].dim_sizes()[0] != 1 {
+        return false;
+    }
+    if operands[2].dim_sizes().to_vec() != vec![1, operands[1].dim_sizes()[1]] {
+        return false;
+    }
+    true
+}
+
+pub fn mult_applies_to_operands(operands: &[TensorSpec<ArmTarget>]) -> bool {
+    operands
+        .iter()
+        .all(|o| o.level() == X86MemoryLevel::RF && o.dim_sizes().iter().all(|&d| d == 1))
+}
```

Comment:
Is all of this duplicate code with x86.rs?

Reply:
That's right, I should have unified these into target. Thank you for pointing this out!